Tests/ApolloTests/Network/URLSessionClientTests.swift

```diff
@@ -153,26 +181,29 @@ class URLSessionClientLiveTests: XCTestCase {
         XCTFail("Task succeeded when it should have been cancelled!")
       }
     }
-    
+
     task.cancel()
-    
+
     self.wait(for: [expectation], timeout: 10)
   }
   
   func testCancellingTaskThroughClientDoesNotCallCompletion() throws {
-    let request = self.request(for: .bytes(count: 102400)) // 102400 is max from HTTPBin
-    
+    let url = URL(string: "http://www.test.com/cancelThroughClient")!
+    let request = request(for: url,
+                          responseData: nil,
+                          statusCode: -1)
+
     let expectation = self.expectation(description: "Cancelled task completed")
     expectation.isInverted = true
     let task = self.client.sendRequest(request) { result in
       // This shouldn't get hit since we cancel the task immediately
       expectation.fulfill()
     }
-    
+
     self.client.cancel(task: task)
-    
+
     self.wait(for: [expectation], timeout: 5)
-    
+
   }
   
   func testMultipleSimultaneousRequests() {
```

Comment:
Ended up not even needing to do some sort of random response delay for this, they complete in a different order each time the test is run.

Reply:
Even if that's the case, it's due to some implementation detail of `URLSession` that is liable to change in the future. I feel like we should be adding some random delay in here anyways, just to ensure we are in control of the behavior of the tests _always_ doing what we expect in the future.

Reply:
Added a random 0-1 seconds execution delay in `MockURLProtocol` to give some randomness to the test responses.